modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -1255,6 +1255,19 @@ void GDScriptAnalyzer::resolve_function_body(GDScriptParser::FunctionNode *p_fun
 	parser->current_function = previous_function;
 }
 
+void GDScriptAnalyzer::resolve_getter_body(GDScriptParser::FunctionNode *p_getter) {
+	resolve_function_body(p_getter);
+
+	GDScriptParser::FunctionNode *previous_function = parser->current_function;
+	parser->current_function = p_getter;
```

Comment:
I don't think a separate method is needed here. The current function context is not used below (lines 1264-1266). Also, in this case, you would also need to switch the static context (see #77151).

You should move lines 1264-1266 inside `resolve_class_body()` after calling `resolve_function_body()` on line 920 and remove the `resolve_getter_body()` method.